### ISSUE TITLE
DHCP options order

### ISSUE THREAD

**Issue by lucianf** [terraform-providers/terraform-provider-oci#829]
Continuation from #821:

DHCP options order is not handled correctly. If you have for example two options:

```hcl
  options {
    type = "DomainNameServer"

    server_type = "VcnLocalPlusInternet"
  }

  options {
    type = "SearchDomain"
    search_domain_names = ["xxx.xxx.oraclevcn.com"]
```

sometimes the provider picks them up in the wrong order from the API and flags it as diff (drift).  You can reproduce the same by switching the order of the options in the TF resource.

I believe this is wrong - the order should not matter, and should not produce a diff.  [The documentation](https://www.terraform.io/docs/providers/oci/r/core_dhcp_options.html) at least does not say anything about this.

**Comment by kilokahn** [comment]
Thanks for reporting this @lucianf! This does look like a case where the diff is unnecessary. In the past we have explicitly made the type of the field in Terraform as a `TypeSet` (instead of the default `TypeList`) where the order doesn't matter. We will follow up with the service team to explicitly confirm on this behavior.

**Comment by afedorch** [comment]
@lucianf the fix was released as part of the 3.37.0. Please give it a try, thanks!
